### PR TITLE
Add LiveObjects read-only public API docs (`ably.d.ts` types)

### PR DESCRIPTION
**Pull request by VeskeR**
`ably.d.ts` types are required to enable LiveObjects package tests in https://github.com/ably/ably-js/pull/1921

## Summary by CodeRabbit

- **New Features**
	- Introduced real-time data structures, including `LiveMap` and `LiveCounter`, for synchronized storage and counters.
	- Added methods for managing updates and subscriptions to live objects.
	- Enhanced TypeScript definitions with new interfaces and types for better usability and functionality.

### CODE REVIEW

File: ably.d.ts [hunk 1]
```diff
@@ -2028,7 +2035,124 @@ export declare interface PushChannel {
...
+   * Retrieves the root {@link LiveMap} object for state on a channel.
+   *
+   * @returns A promise which, upon success, will be fulfilled with a {@link LiveMap} object. Upon failure, the promise will be rejected with an {@link ErrorInfo} object which explains the error.
+   */
+  getRoot(): Promise<LiveMap>;
+}
+
+/**
+ * The `LiveMap` class represents a synchronized key/value storage, similar to a JavaScript Map, where all changes are synchronized across clients in realtime.
+ * Conflict-free resolution for updates follows Last Write Wins (LWW) semantics, meaning that if two clients update the same key in the map, the last change wins.
```
Comment (mschristensen) [inline ably.d.ts:2049]:
```suggestion
 * Conflict-free resolution for updates follows Last Write Wins (LWW) semantics, meaning that if two clients update the same key in the map, the update with the most recent timestamp wins.
```

Comment (mschristensen) [inline ably.d.ts:2049]:
```suggestion
 * Conflict-free resolution for updates follows Last Write Wins (LWW) semantics, meaning that if two clients update the same key in the map, the update with the most recent timestamp wins.
```

File: ably.d.ts [hunk 2]
```diff
@@ -2028,7 +2035,124 @@ export declare interface PushChannel {
...
+  /**
+   * Retrieves the root {@link LiveMap} object for state on a channel.
+   *
+   * @returns A promise which, upon success, will be fulfilled with a {@link LiveMap} object. Upon failure, the promise will be rejected with an {@link ErrorInfo} object which explains the error.
+   */
+  getRoot(): Promise<LiveMap>;
+}
+
+/**
+ * The `LiveMap` class represents a synchronized key/value storage, similar to a JavaScript Map, where all changes are synchronized across clients in realtime.
```
Comment (mschristensen) [inline ably.d.ts:2048]:
```suggestion
 * The `LiveMap` class represents a synchronized key/value map data structure, similar to a JavaScript Map, where all changes are synchronized across clients in realtime.
```

File: ably.d.ts [hunk 3]
```diff
@@ -2028,7 +2035,124 @@ export declare interface PushChannel {
...
+
+/**
+ * Represents a primitive value that can be stored in a {@link LiveMap}.
+ *
+ * For binary data, the resulting type depends on the platform (`Buffer` in Node.js, `ArrayBuffer` elsewhere).
+ */
+export type StateValue = string | number | boolean | Buffer | ArrayBuffer;
+
+/**
+ * The `LiveCounter` class represents a synchronized counter that can be incremented or decremented and is synchronized across clients in realtime.
```
Comment (mschristensen) [inline ably.d.ts:2088]:
```suggestion
 * The `LiveCounter` class represents a counter that can be incremented or decremented and is synchronized across clients in realtime.
```

**Review by mschristensen — APPROVED** [review]
Minor suggestions, but looks good :)

File: ably.d.ts (hunk 2 above)
Comment (VeskeR) [reply ably.d.ts:2048]: I also assume you would like to remove the `synchronized` word from the `synchronized key/value map data structure` string, the same way you suggested for the counter below

Comment (VeskeR) [reply ably.d.ts:2048]:
```suggestion
 * The `LiveMap` class represents a ey/value map data structure, similar to a JavaScript Map, where all changes are synchronized across clients in realtime.
```

**Pull request merged**